arangod/Futures/Future.h

```diff
@@ -0,0 +1,557 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2018 ArangoDB GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Simon Grätzer
+////////////////////////////////////////////////////////////////////////////////
+
+#ifndef ARANGOD_FUTURES_FUTURE_H
+#define ARANGOD_FUTURES_FUTURE_H 1
+
+#include <chrono>
+#include <future>
+#include <thread>
+
+#include "Futures/Exceptions.h"
+#include "Futures/Unit.h"
+#include "Futures/Promise.h"
```

Comment:
`Future.h` includes `Promise.h`

Reply:
But the second time the guard is defined, so it will stop.